### PR TITLE
fix address range functions

### PR DESCRIPTION
**Pull request by invis87**

### CODE REVIEW

**Review by orkunkl — APPROVED** [review]
LGTM. tests for extra coverage would be great if you will like coding more

**Comment by invis87** [comment]
Do not merge please. I will add some tests

**Comment by invis87** [comment]
Tests added. (hope you are OK with that fact that I add `dev-dependency` to wrote tests

**Comment by invis87** [comment]
`cargo fmt` does not change anything. Don't know why `lint` fails

**Comment by maurolacy** [comment]
> `cargo fmt` does not change anything. Don't know why `lint` fails

Likely, some version / rules updates.

Check https://app.circleci.com/pipelines/github/CosmWasm/cosmwasm-plus/1252/workflows/4623eeff-43b1-4e5b-aba5-970584ca9f55/jobs/23104/parallel-runs/0/steps/0-107

**Comment by invis87** [comment]
@maurolacy there is no way to log in in circle without providing `This application will be able to read and write all public and private repository data.` which I do not like.
So, can't open your url.
I did `rustup update`, state:
```
stable-x86_64-apple-darwin updated - rustc 1.53.0 (53cb7b09b 2021-06-17) (from rustc 1.52.1 (9bc8c42bb 2021-05-09))
nightly-x86_64-apple-darwin updated - rustc 1.55.0-nightly (7100b311d 2021-07-01) (from rustc 1.54.0-nightly (1c6868aa2 2021-05-27))
```
Still no changes from `cargo fmt`, so have no idea how to fix.

**Comment by maurolacy** [comment]
> @maurolacy there is no way to log in in circle without providing `This application will be able to read and write all public and private repository data.` which I do not like.
> So, can't open your url.
> I did `rustup update`, state:
> 
> ```
> stable-x86_64-apple-darwin updated - rustc 1.53.0 (53cb7b09b 2021-06-17) (from rustc 1.52.1 (9bc8c42bb 2021-05-09))
> nightly-x86_64-apple-darwin updated - rustc 1.55.0-nightly (7100b311d 2021-07-01) (from rustc 1.54.0-nightly (1c6868aa2 2021-05-27))
> ```
> 
> Still no changes from `cargo fmt`, so have no idea how to fix.

Don't worry, we'll fix it later.

**Comment by maurolacy** [comment]
> > Still no changes from `cargo fmt`, so have no idea how to fix.

You can always try and reproduce the CI setup / tests locally. Take a look a the cicleci config file for reference.

**Comment by invis87** [comment]
`cargo clippy --all-targets -- -D warnings` helped

File: packages/cw0/Cargo.toml [hunk 1]
```diff
@@ -16,3 +16,6 @@ cosmwasm-std = { version = "0.14.0" }
 schemars = "0.8.1"
 serde = { version = "1.0.103", default-features = false, features = ["derive"] }
 thiserror = { version = "1.0.21" }
+
+[dev-dependencies]
+cw-storage-plus = { version = "0.6.1", features = ["iterator"] }
```
Comment (webmaster128) [inline packages/cw0/Cargo.toml:21]: Can you use a local dependency, e.g.

```suggestion
cw-storage-plus = { path = "../../packages/storage-plus", version = "0.6.2", features = ["iterator"] }
```

File: packages/cw0/src/pagination.rs [hunk 2]
```diff
@@ -11,32 +11,20 @@ pub fn maybe_addr(api: &dyn Api, human: Option<String>) -> StdResult<Option<Addr
...
-        Some(human) => {
-            let mut v: Vec<u8> = api.addr_canonicalize(human.as_ref())?.0.into();
-            v.push(0);
-            Ok(Some(v))
-        }
-        None => Ok(None),
-    }
+pub fn calc_range_start(start_after: Option<Addr>) -> Option<Vec<u8>> {
+    start_after.map(|addr| {
+        let mut v: Vec<u8> = addr.as_ref().into();
```
Comment (webmaster128) [inline packages/cw0/src/pagination.rs:16]: There is now `.as_bytes()` which is a bit more explicit and does the same

```suggestion
        let mut v: Vec<u8> = addr.as_bytes().into();
```

File: packages/cw0/src/pagination.rs [hunk 3]
```diff
@@ -11,32 +11,20 @@ pub fn maybe_addr(api: &dyn Api, human: Option<String>) -> StdResult<Option<Addr
...
-pub fn calc_range_end_human(api: &dyn Api, end_before: Option<Addr>) -> StdResult<Option<Vec<u8>>> {
-    match end_before {
-        Some(human) => {
-            let v: Vec<u8> = api.addr_canonicalize(human.as_ref())?.into();
-            Ok(Some(v))
-        }
-        None => Ok(None),
-    }
+pub fn calc_range_end(end_before: Option<Addr>) -> Option<Vec<u8>> {
+    end_before.map(|addr| addr.as_ref().into())
```
Comment (webmaster128) [inline packages/cw0/src/pagination.rs:24]:
```suggestion
    end_before.map(|addr| addr.as_bytes().into())
```

**Review by webmaster128 — COMMENTED** [review]
Just some code style remarks. Mauro and Ethan know the logic better than I do.

File: packages/cw0/src/pagination.rs [hunk 4]
```diff
@@ -11,32 +11,20 @@ pub fn maybe_addr(api: &dyn Api, human: Option<String>) -> StdResult<Option<Addr
...
-) -> StdResult<Option<Vec<u8>>> {
-    match start_after {
-        Some(human) => {
-            let mut v: Vec<u8> = api.addr_canonicalize(human.as_ref())?.0.into();
-            v.push(0);
-            Ok(Some(v))
-        }
-        None => Ok(None),
-    }
+pub fn calc_range_start(start_after: Option<Addr>) -> Option<Vec<u8>> {
```
Comment (maurolacy) [inline packages/cw0/src/pagination.rs:14]: Can only do proper / full reviews beginning next week.

Perhaps the original functions must be kept, as they can be useful for old-style ranges / storage. These new ones can also be suffixed by `_addr` for clarity, but given that this is the new standard, I don't think that's really necessary.

Comment (maurolacy) [reply packages/cw0/src/pagination.rs:14]: I see now that the original argument is `Addr` already so, it's OK to remove / rename these functions.

**Review by maurolacy — APPROVED** [review]
lgtm.

**Comment by invis87** [comment]
```text
  --> packages/cw0/src/pagination.rs:16:35
   |
16 |         let mut v: Vec<u8> = addr.as_bytes().into();
   |                                   ^^^^^^^^ method not found in `Addr`

error[E0599]: no method named `as_bytes` found for struct `Addr` in the current scope
  --> packages/cw0/src/pagination.rs:24:32
   |
24 |     end_before.map(|addr| addr.as_bytes().into())
   |                                ^^^^^^^^ method not found in `Addr`
```

Fix dependency section.

**Comment by webmaster128** [comment]
> ^^^^^^^ method not found in `Addr`

The version of cosmwasm-std is not up to date anymore. Your branch has 0.14. Could you update this PR to the latest main state?

**Comment by invis87** [comment]
> > ^^^^^^^ method not found in `Addr`
> 
> The version of cosmwasm-std is not up to date anymore. Your branch has 0.14. Could you update this PR to the latest main state?

Done, rebase it to `main`

**Comment by orkunkl** [comment]
Merging

**Pull request merged**